omod/src/main/java/org/openmrs/module/pihcore/fragment/controller/dashboardwidgets/PregnancyIndicatorsFragmentController.java

```diff
@@ -0,0 +1,157 @@
+            @SpringBean("encounterService") EncounterService encounterService,
+            @SpringBean EmrApiProperties emrApiProperties,
+            @FragmentParam("app") AppDescriptor app,
+            UiUtils ui,
+            FragmentConfiguration config,
+            FragmentModel model) {
+
+        Object patientConfig = config.get("patient");
+        Patient patient = null;
+        if (patientConfig != null && (patientConfig instanceof PatientDomainWrapper)) {
+            patient = ((PatientDomainWrapper) patientConfig).getPatient();
+        }
+        if (patient == null) {
+            throw new IllegalArgumentException("No patient found. Please pass patient into the configuration");
+        }
+
+        model.put("app", app);
+        model.put("patient", patient);
+        String programUuid = PihCoreUtils.getConfigValue(app, "duringCurrentEnrollmentInProgram");
+
+        Map<String, SimpleObject> fields = new LinkedHashMap<>();
+
+        fields.put("status", SimpleObject.create(
+                "label", "pihcore.status",
+                "css", "red",
+                "obs", ui.message("pihcore.pregnant")
+        ));
+        Concept question = conceptService.getConceptByUuid(PihEmrConfigConstants.CONCEPT_ESTIMATEDDUEDATE_UUID);
+        List<Obs> obsWithinProgram = null;
+        if ( question != null ) {
+            obsWithinProgram = PihCoreUtils.getObsWithinProgram(patient, new HashSet<>(Arrays.asList(question)), null, programUuid);
+        }
+
+        fields.put("dueDate", SimpleObject.create(
+                "label", "pihcore.delivery.dueDate",
+                "css", "red",
+                "obs", (obsWithinProgram!=null && obsWithinProgram.size() > 0 ) ? obsWithinProgram.get(0) : ""
+        ));
+
+        question = conceptService.getConceptByUuid(SierraLeoneConfigConstants.CONCEPT_PREGNANCYRISKFACTORS_UUID);
+        Concept answer = conceptService.getConceptByUuid(PihEmrConfigConstants.CONCEPT_MULTIPLEBIRTH_UUID);
+        if ( question != null && answer != null) {
+            obsWithinProgram = PihCoreUtils.getObsWithinProgram(patient, new HashSet<>(Arrays.asList(question)), new HashSet<>(Arrays.asList(answer)), programUuid);
+        }
+        fields.put("multipleGestation", SimpleObject.create(
+                "label", "pihcore.multiple.birth",
+                "obs", (obsWithinProgram!=null && obsWithinProgram.size() > 0 ) ? ui.message("general.yes") : ui.message("general.no")
+        ));
+
+        question = conceptService.getConceptByUuid(SierraLeoneConfigConstants.CONCEPT_DATETIMEOFBIRTH_UUID);
+        if ( question != null ) {
+            obsWithinProgram = PihCoreUtils.getObsWithinProgram(patient, new HashSet<>(Arrays.asList(question)), null, null);
+        }
+        fields.put("latestDelivery", SimpleObject.create(
+                "label", "pihcore.latest.delivery",
+                "obs", (obsWithinProgram!=null && obsWithinProgram.size() > 0 ) ? obsWithinProgram.get(0) : ""
```

Comment:
Definitely an edge case for twins born a significant time apart, but if there are multiple obs found, shouldn't we take the latest delivery date.

Reply:
Yes, the method we are using here https://github.com/PIH/openmrs-module-pihcore/blob/master/omod/src/main/java/org/openmrs/module/pihcore/PihCoreUtils.java#L53 

actually orders the Obs descending by obsDatetime:
https://github.com/openmrs/openmrs-core/blob/master/api/src/main/java/org/openmrs/api/db/hibernate/HibernateObsDAO.java#L275

But there is a possibility that someone enter manually the second baby born at a sooner date than the first baby? I could edit the code to specifically look at the value of the obs and get the most recent obs.value?